caom2-search-server/src/main/resources/META-INF/resources/js/cadc.search.app.js

```diff
@@ -1066,62 +1066,152 @@
+                  // a JSON blob to transmit tuples built below
+
+                  // build cutout DALI string
+                  var $nextPlaneCutout = 'CIRCLE ' + $nextRow['caom2:Upload.ra'] + " " + $nextRow['caom2:Upload.dec']
+                    + ' ' + $nextRow['caom2:Upload.radius']
+
+                  // grab target name for label
+                  var $nextPlaneTargetName = $nextRow['caom2:Upload.target']
+
+                  var tuple = {
+                    "tupleID" : $nextPlaneURI,
+                    "shape" : $nextPlaneCutout,
+                    "label" : $nextPlaneTargetName
+                  }
+
+                  downloadTuples.push(tuple)
 
-                downloadForm.append($input)
+                } else {
+                  // hidden input used with resolver form
+                  // downloadManager request will have URI=#&URI=# format
+                  var $input = $('<input>')
+                  $input.prop('type', 'hidden')
+                  $input.prop('name', 'uri')
+                  $input.prop('id', $nextPlaneURI)
+                  $input.val($nextPlaneURI)
+
+                  downloadForm.append($input)
+                }
               }
 
               // Story 1566, when all 'Product Types'
               // checkboxes are checked, do not send any
+              // both (2739) - although I wonder if the 'all checked' check
+              //being done twice is necessary (after the submit, for example.)
               var allChecked =
-                      downloadForm
-                          .find('input.product_type_option_flag')
-                          .not(':checked').length === 0
+                    downloadForm
+                      .find('input.product_type_option_flag')
+                      .not(':checked').length === 0
               if (allChecked) {
                 // disable all 'Product Types' checkboxes
                 $.each(
-                    downloadForm.find('input.product_type_option_flag:checked'),
-                    function () {
-                      $(this).prop('disabled', true)
-                    }
+                  downloadForm.find('input.product_type_option_flag:checked'),
+                  function () {
+                    $(this).prop('disabled', true)
+                  }
                 )
               }
 
-              window.open('', 'DOWNLOAD', '')
-              downloadForm.submit()
+              // Now get down to submitting the data
+              if (fromInputFile) {
+                // iterate through downloadTuples and make the badgerfish json
+                var badgerfishTuples = "";
+                for (i=0; i<downloadTuples.length; i++) {
+                  var tupleJSON = "{\"tuple\":{\"tupleID\":{\"$\":\"" + downloadTuples[i].tupleID + "\"}," +
+                    "\"shape\":{\"$\":\"" + downloadTuples[i].shape + "\"}," +
+                    "\"label\":{\"$\":\"" + downloadTuples[i].label + "\"}}}";
```

Comment:
This is better suited to be actual JSON.
```javascript
var badgerfishTuples = {}
...
var tupleObject = {}
tupleObject.tuple = {}
tupleObject.tuple.tupleID = {"$": downloadTuples[i].tupleID}
tupleObject.tuple.shape = ...
...
```
Pure JSON is easier to read and doesn't involve escaping over and over again.

Reply:
done.